cmake/RippledBase.cmake

```diff
@@ -0,0 +1,21 @@
+set (CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Builds/CMake ${CMAKE_MODULE_PATH})
+include (CheckCXXCompilerFlag)
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.11)
+  include (FetchContent)
+endif ()
+if (MSVC AND CMAKE_VERSION VERSION_LESS 3.12)
+  message (FATAL_ERROR "MSVC requires cmake 3.12 or greater for proper boost support")
+endif ()
+include (ExternalProject)
+include (CMakeFuncs) # must come *after* ExternalProject b/c it overrides one function in EP
+include (ProcessorCount)
+if (target)
+  message (WARNING
+    "The target option is deprecated and will be removed in a future release")
+  parse_target()
+endif ()
+project (rippled)
```

Comment:
project declarations need to be in the top level cmake file I believe:  https://cmake.org/cmake/help/latest/command/project.html